Mathlib/CategoryTheory/Limits/Sifted.lean

```diff
@@ -101,4 +111,224 @@ end IsSifted
 
 end
 
+noncomputable section SmallCategory
+
+open MonoidalCategory ChosenFiniteProducts
+
+namespace IsSifted
+
+variable {C : Type u} [SmallCategory.{u} C]
+
+/-- We introducte an auxiliary external product on presheaves for convenience. -/
+@[simps!]
+private def externalProductFunctor : ((C ⥤ Type u) × (C ⥤ Type u) ⥤ C × C ⥤ Type u) :=
+  prodFunctor ⋙ (whiskeringRight _ _ _).obj (tensor _)
+
+private abbrev externalProduct (X Y : (C ⥤ Type u)) : (C × C ⥤ Type u) :=
+  externalProductFunctor.obj <| Prod.mk X Y
+
+local infixr:80 " ⊠ " => externalProduct
+
+section
+
+variable (X Y : C ⥤ Type u)
+/-- An auxiliary isomorphisms that computes the colimit of a functor `C × C ⥤ Type`
+that decomposes as an external product of two functors `C ⥤ Type` -/
+private def colimBoxIsoColimTensColim : colimit (X ⊠ Y) ≅ (colimit X) ⊗ (colimit Y) :=
+  calc colimit (X ⊠ Y)
+    _ ≅ colimit <| (curry.obj _) ⋙ colim := colimitIsoColimitCurryCompColim _
+    _ ≅ colimit <| _ ⋙ colim := HasColimit.isoOfNatIso <| isoWhiskerRight
+        (NatIso.ofComponents (fun _ ↦ NatIso.ofComponents (fun _ ↦ β_ _ _)) :
```

Comment:
I suspect that given different roles to the two variables, it should not be necessary to use the braiding. (With the `PreservesLimit₂` API I am suggesting, I would think that `PreservesLimit₂ K₁ K₂ G ↔PreservesLimit₂ K₂ K₁ G.flip` should always hold.)

Reply:
(Actually, we may have to use the braiding in order to get the result for the tensor product in a monoidal closed category, but I believe this should be phrased first in terms of the commutation of both `tensorLeft` and `tensorRight`, and in the braided case, we shall deduce the latter from the former.)